File: Meets/Controllers/api/dto/Meeting/MeetingRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Meets.Controllers.api.dto.Meeting
{
    /// <summary>
    /// Приглашение на встречу
    /// </summary>
    public class MeetingRequest
    {
        [Required]
        public ulong TargetId { get; set; }

        [Required]
        public DateTime MeetingDate { get; set; }

        [Required]
        public bool IsOnline { get; set; }

        /// <summary>
        /// Место встречи. Может быть адресом или вводится в свободной форме
        /// </summary>
        [Required]
        public string Place { get; set; }

        [Required]
        public string Message { get; set; }
    }
}

File: Meets/ClientApp/src/contracts/meeting/MeetingRequest.ts
import UserCardResponse from "../user/UserCardResponse";
import UserDTO from "../user/UserDTO";
import UserListItemDTO from "../user/UserListItemDTO";

export default class MeetingRequest{

    static create(user: UserCardResponse | UserListItemDTO, message: string): MeetingRequest {

        let res = new MeetingRequest();

        res.targetId = user.id;
        res.message = message;

        return res;
    }


    targetId: number

    meetingDate: any

    isOnline: boolean = false

    place: string = ''

    message: string = ''
}